Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/FRFS/OnConfirm.hs

```diff
@@ -49,20 +58,74 @@ validateRequest DOrder {..} = do
       void $ QTBooking.updateBPPOrderIdAndStatusById (Just bppOrderId) Booking.FAILED booking.id
-      callBPPCancel booking bapConfig
+      let updatedBooking = booking {Booking.bppOrderId = Just bppOrderId}
+      callBPPCancel updatedBooking bapConfig
       throwM $ (InvalidRequest "Booking expired, initated cancel request")
     else do
       -- Booking is valid, proceed to onConfirm handler
       merchant <- QMerch.findById merchantId >>= fromMaybeM (MerchantNotFound merchantId.getId)
       return (merchant, booking)
 
 onConfirm :: Merchant -> Booking.FRFSTicketBooking -> DOrder -> Flow ()
-onConfirm _merchant booking dOrder = do
+onConfirm merchant booking' dOrder = do
+  let booking = booking' {Booking.bppOrderId = Just dOrder.bppOrderId}
   tickets <- traverse (mkTicket booking) dOrder.tickets
   void $ QTicket.createMany tickets
   void $ QTBooking.updateBPPOrderIdAndStatusById (Just dOrder.bppOrderId) Booking.CONFIRMED booking.id
+  buildReconTable merchant booking dOrder tickets
   return ()
 
+buildReconTable :: Merchant -> Booking.FRFSTicketBooking -> DOrder -> [Ticket.FRFSTicket] -> Flow ()
+buildReconTable merchant booking _dOrder tickets = do
+  bapConfig <- QBC.findByMerchantIdDomainAndVehicle (Just merchant.id) (show Spec.FRFS) METRO >>= fromMaybeM (InternalError "Beckn Config not found")
+  quote <- runInReplica $ QFRFSQuote.findById booking.quoteId >>= fromMaybeM (QuoteNotFound booking.quoteId.getId)
+  fromStation <- runInReplica $ QStation.findById booking.fromStationId >>= fromMaybeM (InternalError "Station not found")
+  toStation <- runInReplica $ QStation.findById booking.toStationId >>= fromMaybeM (InternalError "Station not found")
+  person <- runInReplica $ QPerson.findById booking.riderId >>= fromMaybeM (PersonNotFound booking.riderId.getId)
+  transactionRefNumber <- booking.paymentTxnId & fromMaybeM (InternalError "Payment Txn Id not found in booking")
+  paymentBooking <- runInReplica $ QFRFSTicketBookingPayment.findNewTBPByBookingId booking.id >>= fromMaybeM (InvalidRequest "Payment booking not found for approved TicketBookingId")
+  txn <- runInReplica $ QPaymentTransaction.findNewTransactionByOrderId paymentBooking.paymentOrderId >>= fromMaybeM (InvalidRequest "Payment booking not found for approved TicketBookingId")
+  mRiderNumber <- mapM decrypt person.mobileNumber
+  now <- getCurrentTime
+  bppOrderId <- booking.bppOrderId & fromMaybeM (InternalError "BPP Order Id not found in booking")
+  let totalOrderValue = booking.price
+  let finderFee :: HighPrecMoney = fromMaybe 0 $ (readMaybe . T.unpack) =<< bapConfig.buyerFinderFee
+  let reconEntry =
+        Recon.FRFSRecon
+          { Recon.id = "",
+            Recon.beneficiaryIFSC = booking.bppBankCode,
+            Recon.beneficiaryBankAccount = booking.bppBankAccountNumber,
+            Recon.buyerFinderFee = finderFee,
+            Recon.collectorIFSC = bapConfig.bapIFSC,
```

Comment:
Move BAP bankAccNo also to `BapConfig` table and remove `BknPaymentParams` type.

Reply:
Let's fix this in next PR @RitikaHanish @VD321